library/core/src/ffi/c_str.rs

```diff
@@ -296,6 +317,12 @@ impl CStr {
     #[must_use]
     #[stable(feature = "rust1", since = "1.0.0")]
     #[rustc_const_stable(feature = "const_cstr_from_ptr", since = "1.81.0")]
+    #[requires(!ptr.is_null() && is_null_terminated(ptr))]
+    #[ensures(|result: &&CStr| {
+        result.is_safe() && 
+        core::ptr::eq(result.as_ptr(), ptr) && 
+        unsafe { result.count_bytes() == strlen(ptr) }
```

Comment:
let me know if these 2 checks are required. I feel only `#[ensures(|result: &&CStr| result.is_safe())]` should be good @celinval

Reply:
Just checking safety is enough. Thanks!

Reply:
Thanks @celinval! fixed it.